utils/interface-validator.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2024 Canonical
+# See LICENSE file for licensing details.
+
+"""Schema validator for `interface.yaml` files."""
+import yaml
+import glob
+import re 
```

Comment:
* pyyaml isn't a dep in tox.ini, is it just being pulled in via something else (pydantic?)? It would be better if it was explicitly listed since it's explicitly imported.
* Does this repo have a formatter that includes imports? These would normally be:

```suggestion
import glob
import re

import yaml
```

ie. sections for stdlib and third-party, and in alphabetical order. But maybe that's not the style here?

Similar with the `from x import y` ones below.

Reply:
> pyyaml isn't a dep in tox.ini, is it just being pulled in via something else (pydantic?)? It would be better if it was explicitly listed since it's explicitly imported.

it is :) https://github.com/canonical/charm-relation-interfaces/blob/c4145711cd57c1f34077a4852d9fd0570f1752aa/tox.ini#L88